pkg/reconciler/apis/apibinding/apibinding_controller.go

```diff
@@ -119,8 +121,8 @@ func NewController(
 			return apiResourceSchema, err
 		},
 
-		createCRD: func(ctx context.Context, clusterName logicalcluster.Name, crd *apiextensionsv1.CustomResourceDefinition) (*apiextensionsv1.CustomResourceDefinition, error) {
```

Comment:
@ncdc unclear how this has evolved but it seemed weird to pass in an object that *has* a cluster, then ignore it and use a different value instead

Reply:
please don't use this pattern. **Always** pass the cluster name and always ignore the value in the object. This kind of code here become unreadable in non-trivial cases.

Reply:
Why? It seems entirely non-obvious and confusing to explicitly set a cluster in the object, then ignore it